Core/src/org/sleuthkit/autopsy/report/GeneralReportModule.java

```diff
@@ -25,10 +25,36 @@
      * at the path specified and updating progress via the progressPanel object.
      *
      * @param baseReportDir Base directory that reports are being stored in.
-     *                      Report should go into baseReportDir +
-     *                      getRelativeFilePath().
+     * Report should go into baseReportDir + getRelativeFilePath().
      * @param progressPanel panel to update the report's progress with
      */
     public void generateReport(String baseReportDir, ReportProgressPanel progressPanel);
 
+    /**
+     * Determines if the module supports report generation on a subset of data
+     * sources in a case. Defaults to false. Modules that override this to true
+     * should implement all generateReport overloads. The data source selections
+     * are stored in the GeneralReportSettings instance.
```

Comment:
"Modules that override this to true should implement all generateReport overloads."

See comments in Jira.